go/client/cmd_wallet_cancel_all.go

```diff
@@ -0,0 +1,196 @@
+	}
+
+	file, err := os.Open(c.outfile)
+	if err != nil {
+		return out, err
+	}
+	defer file.Close()
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		row := scanner.Text()
+		out[row] = true
+	}
+	if err := scanner.Err(); err != nil {
+		return out, err
+	}
+	return out, nil
+}
+
+func (c *cmdWalletCancelAll) Run() (err error) {
+	defer transformStellarCLIError(&err)
+	dui := c.G().UI.GetDumbOutputUI()
+	cli, err := GetWalletClient(c.G())
+	if err != nil {
+		return err
+	}
+
+	out, err := c.getOutTxIDs()
+	if err != nil {
+		return err
+	}
+	in, err := c.getInTxIDs(out)
+	if err != nil {
+		return err
+	}
+
+	var outfile *os.File = nil
+	// if c.outfile defined, append finished txIDs
+	if len(c.outfile) > 0 {
+		outfile, err = os.OpenFile("out.txt",
+			os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+		if err != nil {
+			return err
+		}
+		defer outfile.Close()
+	} else {
+		dui.Printf("Note: no outfile specified, so not keeping track of complete txIDs.\n")
+	}
+
+	maxWorkers := 100 // OR runtime.GOMAXPROCS(0); goal is to not overwhelm stellarnet
+	sem := semaphore.NewWeighted(int64(maxWorkers))
+	dui.Printf("Canceling %d txIDs with %d maxWorkers; %d txIDs already complete...\n", len(in), maxWorkers, len(out))
+
+	var wg sync.WaitGroup
+	successCount := 0
+	skipCount := 0
+	failCount := 0
+	writeErrorCount := 0
+	for _, txID := range in {
+		if err := sem.Acquire(context.TODO(), 1); err != nil {
+			// don't expect to get here
+			log.Printf("Failed to acquire semaphore: %v", err)
+			return nil
+		}
+		wg.Add(1)
+		go func(txID string) {
+			defer func() {
+				sem.Release(1)
+				wg.Done()
+			}()
+			res, err := cli.ClaimCLILocal(context.TODO(), stellar1.ClaimCLILocalArg{TxID: txID})
+			if err != nil {
+				// you may want to add more cases for when an erroring txID should be
+				// added to out.txt
+				if strings.Contains(err.Error(), "Payment already claimed by ") {
+					dui.Printf("SKIP: %+v, %+v\n", txID, err.Error())
+					skipCount++
+					if outfile != nil {
+						if _, err := outfile.WriteString(fmt.Sprintf("%+v\n", txID)); err != nil {
+							// would get picked up to attempt cancel on the next run...
+							fmt.Printf("WRITE ERROR, %+v, %+v\n", txID, err.Error())
+							writeErrorCount++
+						}
+					}
+				} else {
+					dui.Printf("ERROR: %+v, %+v\n", txID, err.Error())
+					failCount++
+				}
+			} else {
+				dui.Printf("SUCCESS: %+v, %+v\n", txID, res)
+				successCount++
+				if outfile != nil {
+					if _, err := outfile.WriteString(fmt.Sprintf("%+v\n", txID)); err != nil {
+						// would get picked up to attempt cancel on the next run...
+						fmt.Printf("WRITE ERROR, %+v, %+v\n", txID, err.Error())
+						writeErrorCount++
+					}
+				}
+			}
+		}(txID)
+	}
+
+	wg.Wait()
+	dui.Printf("Finished canceling. successCount=%d, skipCount=%d, failCount=%d, writeErrorCount=%d\n\n", successCount, skipCount, failCount, writeErrorCount)
+	return nil
+}
+
+func (c *cmdWalletCancelAll) GetUsage() libkb.Usage {
```

Comment:
is this how to unlist the cmd?

Reply:
Use `Unlisted: true` like [this](https://github.com/keybase/client/blob/master/go/client/cmd_account_reset_cancel.go#L21).

To be honest I don't know what `GetUsage` is for, but probably copy what [cmdWalletCancel does](https://github.com/keybase/client/blob/master/go/client/cmd_wallet_cancel.go#L63:L63)

Reply:
done